pennylane/operation.py

```diff
@@ -559,28 +558,67 @@ def hyperparameters(self):
         self._hyperparameters = {}
         return self._hyperparameters
 
-    @staticmethod
-    def decomposition(*params, wires):
-        """Defines a decomposition of this operator into products of other operators.
+    def decomposition(self):
+        """The decomposition of the Operator into a product of more fundamental gates.
 
-        Args:
-            params (tuple[float, int, array]): operator parameters
-            wires (Union(Sequence[int], Wires)): wires the operator acts on
+        .. math:: O = O_1 O_2 \dots O_n
+
+        Returns ``None`` is this Operator does not define its decomposition.
+
+        .. note::
+            By default, this method calls the static method ``compute_decomposition``.
+            Unless the ``compute_decomposition`` has a custom signature, this method should
+            not be overwritten.
 
         Returns:
-            list[Operation]
+            tuple[Operator]: The decomposition of the Operator into lower level operations
+
+        **Example:**
+
+        >>> qml.IsingXX(1.23, wires=(0,1)).decomposition()
+        (CNOT(wires=[0, 1]), RX(1.23, wires=[0]), CNOT(wires=[0, 1]))
+
         """
-        raise NotImplementedError
+        return self.compute_decomposition(*self.parameters, self.wires, **self.hyperparameters)
+
+    @staticmethod
+    def compute_decomposition(*params, wires=None, **hyperparameters):
+        """Determine the Operator's decomposition for specified parameters, wires,
+        and hyperparameters. The decomposition defines an Operator as a product of
+        more fundamental gates:
 
-    def decompose(self):
-        """Decomposes this operator into products of other operators.
+        .. math:: O = O_1 O_2 \dots O_n.
+
+        ``compute_decomposition`` is a static method and can provide the decomposition of a given
+        operator without creating a specific instance. The instance method ``decomposition`` uses
+        this method and the instance's variables.
+
+        Returns ``None`` is this Operator does not define its decomposition.
+
+        .. note::
+            This method gets overwritten by subclasses and helps define the ``decomposition`` and
+            ``expand`` methods. By default, this method should always take the Operator's
+            parameters, wires, and hyperparameters as inputs, even if the decomposition is
+            independent of these values.
+
+        Args:
+            *params: Variable length argument list.  Should match the ``parameters`` attribute
+
+        Keyword Args:
+            wires (Iterable[Number, str], Number, str, Wires): Wires that the operator acts on.
+            **hyperparameters: Variable length keyword arguments.  Should match the
+            ``hyperparameters`` attribute.
```

Comment:
I think we don't distinguish Keyword Args and Args anywhere else?

Reply:
In the base class, wires and hyperparameters **must** be passed as keyword arguments, because `*args` can take any length. If wires or hyperparameters are passed as positional arguments, then they are treated as one of the parameters.

This distinction doesn't apply when we override the call signatures in the child classes, because then the size of the parameters is explicitly specified.